src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -206,23 +206,69 @@ int enroll_tpm2(struct crypt_device *cd,
                         return log_debug_errno(r, "Failed to read TPM PCR signature: %m");
         }
 
-        r = tpm2_seal(device,
-                      hash_pcr_mask,
-                      pubkey, pubkey_size,
-                      pubkey_pcr_mask,
+        _cleanup_(tpm2_context_unrefp) Tpm2Context *tpm2_context = NULL;
+        r = tpm2_context_new(device, &tpm2_context);
+        if (r < 0)
+                return r;
+
+        bool pcr_value_specified = false;
+        for (size_t i = 0; i < n_hash_pcr_values; i++)
+                if (hash_pcr_values[i].value.size > 0) {
+                        pcr_value_specified = true;
+                        break;
+                }
```

Comment:
mabe make this loop a helper function of its own?

Reply:
ack, added `tpm2_pcr_values_has_any_values()` and `tpm2_pcr_values_has_all_values()`